### PR TITLE
Add `tarballProxyEndpoints` workbench option

### PR DESCRIPTION
**Pull request by joyceerhl**
As discussed @bpasero and @JacksonKearl --embedders can set `tarballProxyEndpoints` in options, and extensions can read this value using `workbench.getTarballProxyEndpoints`.

Please let me know if there is a better place to register the `workbench.getTarballProxyEndpoints` command.

### CODE REVIEW

File: src/vs/workbench/workbench.web.api.ts [hunk 1]
```diff
@@ -412,6 +412,12 @@ interface IWorkbenchConstructionOptions {
 	 */
 	readonly codeExchangeProxyEndpoints?: { [providerId: string]: string }
 
+	/**
+	 * [TEMPORARY]: This will be removed soon.
+	 * Endpoints to be used for proxying repository tarball download calls in the browser.
+	 */
+	readonly tarballProxyEndpoints?: { [providerId: string]: string }
```
Comment (bpasero) [inline src/vs/workbench/workbench.web.api.ts:419]: Please create a follow up issue to keep in mind to remove this API again.

File: src/vs/workbench/services/environment/browser/environmentService.ts [hunk 2]
```diff
@@ -18,13 +18,16 @@ import { parseLineAndColumnAware } from 'vs/base/common/extpath';
...
+import { CommandsRegistry } from 'vs/platform/commands/common/commands';
 
 class BrowserWorkbenchConfiguration implements IWindowConfiguration {
 
 	constructor(
 		private readonly options: IBrowserWorkbenchOptions,
 		private readonly payload: Map<string, string> | undefined
-	) { }
+	) {
+		CommandsRegistry.registerCommand('workbench.getTarballProxyEndpoints', () => (this.options.tarballProxyEndpoints ?? {}));
```
Comment (bpasero) [inline src/vs/workbench/services/environment/browser/environmentService.ts:29]: I would rather register the command in `workbench.web.api.ts` where also the API exists, maybe from the `create` method. We already use `CommandRegistry` there.

**Review by bpasero — CHANGES_REQUESTED** [review]

File: src/vs/workbench/workbench.web.api.ts (hunk 1 above)
Comment (joaomoreno) [reply src/vs/workbench/workbench.web.api.ts:419]: Maybe we also prefix with an `_`?

File: src/vs/workbench/services/environment/browser/environmentService.ts (hunk 2 above)
Comment (joaomoreno) [reply src/vs/workbench/services/environment/browser/environmentService.ts:29]: Also should it rather be called `_workbench.getTarballProxyEndpoints`?

File: src/vs/workbench/workbench.web.api.ts (hunk 1 above)
Comment (joyceerhl) [reply src/vs/workbench/workbench.web.api.ts:419]: @bpasero filed https://github.com/microsoft/vscode/issues/138335. Should I include that in the comment too?

@joaomoreno is that naming convention done for internally-consumed commands, or for commands that are proposed and subject to removal?

Comment (bpasero) [reply src/vs/workbench/workbench.web.api.ts:419]: No I think a back reference is not needed.

**Review by bpasero — APPROVED** [review]

**Review by joaomoreno — APPROVED** [review]

**Pull request merged**